src/PhpParser/NodeTraverser/RectorNodeTraverser.php

```diff
@@ -42,10 +49,36 @@ public function refreshPhpRectors(array $rectors): void
     {
         $this->rectors = $rectors;
         $this->visitors = [];
+        $this->visitorsPerNodeClass = [];
 
         $this->areNodeVisitorsPrepared = false;
     }
 
+    /**
+     * We return the list of visitors (rector rules) that can be applied to each node class
+     * This list is cached so that we don't need to continually check if a rule can be applied to a node
+     *
+     * @return NodeVisitor[]
+     */
+    public function getVisitorsForNode(Node $node): array
+    {
+        $nodeClass = $node::class;
+        if (! isset($this->visitorsPerNodeClass[$nodeClass])) {
+            $this->visitorsPerNodeClass[$nodeClass] = [];
+            foreach ($this->visitors as $visitor) {
+                assert($visitor instanceof RectorInterface);
+                foreach ($visitor->getNodeTypes() as $nodeType) {
+                    if (is_a($nodeClass, $nodeType, true)) {
+                        $this->visitorsPerNodeClass[$nodeClass][] = $visitor;
+                        break;
```

Comment:
I prefer to use `continue 2;` instead for easier understanding which loop to continue for future first read code:

```suggestion
                        continue 2;
```

so we know it continue to next outer loop, also show that there is no other functionality after inner loop ends as it continue to outer loop already.

Reply:
I see, that just preferable, reading "break" on the first time looks like it gonna stop, but it actually not as it on inner loop, continue 2 make clear there is no other statement executed after inner loop if it hit, eg:

```php
foreach ($outer as $o) {

    foreach ($o as $i) {
         break;
    }
    
    echo "this executed with break";
}
```

with `continue 2`, if it hit, the "echo" not shown

Reply:
I understand what you mean, but in this case there are no statements after the inner loop so `break` and `continue 2` are equivalent. And I still feel that `break` feels more intuitive here.

Reply:
we use continue 2 in code base so I prefer continue 2 for consistency usage :) https://github.com/search?q=repo%3Arectorphp%2Frector-src+%22continue+2%22&type=code

break is for only break current loop, but allow execution after it, otherwise, use continue 2 to directly continue to outer loop, also, using continue 2 have less number of operation as it less branch to check

Reply:
Done!